packages/core/stellar-sdk/src/utils/stellar-utils.ts

```
@@ -175,3 +189,87 @@ export const isValidAssetCode = (assetCode: string): boolean => {
+/**
+ * Validates a Stellar wallet memo.
+ * @param memo - Memo to validate
+ * @throws Error if memo is invalid
+ */
+export function validateMemo(memo: string): void {
+  if (!memo || typeof memo !== 'string') {
+    throw new Error('Memo must be a non-empty string');
+  }
+
+  // Reject control characters
+  if (/[\x00-\x08\x0B-\x0C\x0E-\x1F\x7F]/.test(memo)) {
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)